transformer_engine/pytorch/attention.py

```diff
@@ -570,14 +575,15 @@ def forward(
             for x in [query_layer, key_layer, value_layer]
         ]
 
+        global _cu_seqlens_q, _cu_seqlens_kv, _indices_q, _indices_kv
```

Comment:
Why do we need these to be global variables? They don't seem to be used outside this function.

Reply:
They're only calculated for `layer_number==1` if layer number is provided by the user and then cached for the other layers (as an optimization). Since it's just the cumulative sequence lengths the extra memory consumption is not an issue. @timmoon10

Reply:
This seems delicate, but I suppose `layer_number` should be considered an advanced optimization.